From c963186769162285b7ff32a80863e4104119dd74 Mon Sep 17 00:00:00 2001
From: Xiao Chen <xiao@apache.org>
Date: Thu, 23 Nov 2017 09:00:59 -0800
Subject: [PATCH 2699/2748] HADOOP-15067. GC time percentage reported in
 JvmMetrics should be a gauge, not counter.
 Contributed by Misha Dmitriev.

(cherry picked from commit d162252d7a7223631ff66ba0210953296407e55f)
(cherry picked from commit a4674c3e4085dbd5bd3a62693bfd4c479143609c)

Change-Id: Ieb65488af89a61e82f30dd056b8d02774e376aa7
---
 .../apache/hadoop/metrics2/source/JvmMetrics.java  |    2 +-
 .../hadoop/metrics2/source/TestJvmMetrics.java     |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/source/JvmMetrics.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/source/JvmMetrics.java
index 31eb7c7..e810d33 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/source/JvmMetrics.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/source/JvmMetrics.java
@@ -175,7 +175,7 @@ private void getGcUsage(MetricsRecordBuilder rb) {
     }
 
     if (gcTimeMonitor != null) {
-      rb.addCounter(GcTimePercentage,
+      rb.addGauge(GcTimePercentage,
           gcTimeMonitor.getLatestGcData().getGcTimePercentage());
     }
   }
diff --git a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/metrics2/source/TestJvmMetrics.java b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/metrics2/source/TestJvmMetrics.java
index 5320b6e..aa1b009 100644
--- a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/metrics2/source/TestJvmMetrics.java
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/metrics2/source/TestJvmMetrics.java
@@ -101,7 +101,7 @@ public void testGcTimeMonitorPresence() {
     verify(rb).tag(SessionId, "test");
     for (JvmMetricsInfo info : JvmMetricsInfo.values()) {
       if (info.name().equals("GcTimePercentage")) {
-        verify(rb).addCounter(eq(info), anyInt());
+        verify(rb).addGauge(eq(info), anyInt());
       }
     }
   }
-- 
1.7.9.5

